File: DislinktFrontApp/src/app/model/post.ts
import { Media } from "./media";
import { Reaction } from "./reaction";

export class Post{
    postId: string;
	userId: string;
	creationTime:any;
	mediaAssets: Array<Media>
	postText:string;
	reactions:Array<Reaction>
	comments:Array<Comment>

    constructor(){

            this.postId='';
            this.userId='';
            this.creationTime=null;
            this.mediaAssets=[];
            this.postText='';
            this.reactions=[];
            this.comments=[];

    }

    
}

File: post/model/post.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Post struct {
	PostID       primitive.ObjectID `bson:"_id" json:"postId"`
	UserEmail    string             `bson:"userEmail" json:"userEmail"`
	CreationTime time.Time          `json:"creationTime"`
	MediaAssets  []Media            `json:"mediaAssets"`
	PostText     string             `json:"postText"`
	Reactions    []Reaction         `json:"reactions"`
	Comments     []Comment          `json:"comments"`
}
